addons/web_editor/static/lib/odoo-editor/src/commands/commands.js

```diff
@@ -98,12 +98,14 @@ function insert(editor, data, isText = true) {
         fakeEl.replaceChildren(...p.childNodes);
     } else if (fakeEl.childElementCount > 1) {
         // Grab the content of the first child block and isolate it.
-        if (isBlock(fakeEl.firstChild) && !['TABLE', 'UL', 'OL'].includes(fakeEl.firstChild.nodeName)) {
+        if (isBlock(fakeEl.firstChild) &&
+            !['TABLE', 'UL', 'OL'].includes(fakeEl.firstChild.nodeName) && startNode.textContent !== '' || fakeEl.firstChild.nodeName === 'P') {
```

Comment:
keep them separated if condition is lengthy for better readability.
if (isBlock(fakeEl.firstChild) &&
            (!['TABLE', 'UL', 'OL'].includes(fakeEl.firstChild.nodeName) &&
             startNode.textContent !== '' ) || 
             fakeEl.firstChild.nodeName === 'P') {...}